src/journeys_utils.js

```diff
@@ -908,3 +908,55 @@ journeys_utils.setJourneyLinkData = function(linkData) {
 	}
 	journeys_utils.journeyLinkData = data;
 };
+
+journeys_utils.getValueForKeyInBranchViewData = function(key) {
+	if(!journeys_utils){
+		return false;
+	}
+
+	if(!journeys_utils.branch){
+		return false;
+	}
+
+	if(!journeys_utils.branch._branchViewData){
+		return false;
+	}
+
+	if(!journeys_utils.branch._branchViewData.data){
+		return false;
+	}
+
+	return journeys_utils.branch._branchViewData.data[key];
+};
+
+journeys_utils.hasJourneyCtaLink = function () {
+	if(!journeys_utils.getValueForKeyInBranchViewData('$journeys_cta')){
+		return false;
+	}
+
+	return journeys_utils.getBranchViewDataItemOrUndefined('$journeys_cta').length > 0;
+};
+
+journeys_utils.getBranchViewDataItemOrUndefined = function(name){
+	if(journeys_utils.getValueForKeyInBranchViewData(name)){
+		return journeys_utils.branch._branchViewData.data[name];
+	}
+	 return undefined;
+};
+
+journeys_utils.getJourneyCtaLink = function () {
+	return journeys_utils.getBranchViewDataItemOrUndefined('$journeys_cta');
+};
+
+journeys_utils.tryReplaceJourneyCtaLink = function (html){
```

Comment:
Can we call this replaceJourneyCtaLink(link, html), and we can check if link exists in the function:

if (!link) { return html; } 

... do other stuff

Reply:
or detectAndReplaceCustomCTALink()

PS this can be done in the backend as well:

https://github.com/BranchMetrics/branch-backend/blob/master/branch/branchview.js#L266

Reply:
detectAndReplaceWithCustomCTALink. The function name shows us that we are not doing the Single Responsibility Principle in that function, which is not good.

Reply:
But you are right a small refactoring could be an improvement.